web/src/test/java/com/microsoft/applicationinsights/web/internal/httputils/HttpServerHandlerTest.java

```diff
@@ -0,0 +1,171 @@
+package com.microsoft.applicationinsights.web.internal.httputils;
+
+import com.microsoft.applicationinsights.TelemetryClient;
+import com.microsoft.applicationinsights.TelemetryConfiguration;
+import com.microsoft.applicationinsights.telemetry.RequestTelemetry;
+import com.microsoft.applicationinsights.web.internal.RequestTelemetryContext;
+import com.microsoft.applicationinsights.web.internal.ThreadContext;
+import com.microsoft.applicationinsights.web.internal.WebModulesContainer;
+import org.hamcrest.CoreMatchers;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+import org.mockito.InOrder;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+import org.mockito.Spy;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.net.MalformedURLException;
+import java.util.Date;
+
+import static org.hamcrest.CoreMatchers.*;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.core.Is.is;
+import static org.hamcrest.core.StringContains.containsString;
+import static org.mockito.Mockito.*;
+
+@RunWith(JUnit4.class)
+public class HttpServerHandlerTest {
+
+    @Rule public final ExpectedException thrown = ExpectedException.none();
+    @Mock public HttpExtractor<HttpServletRequest, HttpServletResponse> extractor;
+    private TelemetryConfiguration telemetryConfiguration = TelemetryConfiguration.getActive();
+    @Spy public WebModulesContainer<HttpServletRequest,HttpServletResponse> webModulesContainer =
+        new WebModulesContainer<>(telemetryConfiguration);
+    @Spy public TelemetryClient telemetryClient;
+    @InjectMocks HttpServerHandler<HttpServletRequest, HttpServletResponse> httpServerHandler;
+    @Mock HttpServletRequest request;
+    @Mock HttpServletResponse response;
+    @Mock HttpServletRequest requestWithQueryString;
+    private String url = "http://www.abc.com/xyz/opq";
+    private String url1 = "http://30thh.loc:8480/app/test%3F/a%3F+b;jsessionid=S%3F+ID?p+1=c+d&p+2=e+f#a";
```

Comment:
Do we want to cover IP address as the host? What about IPv6?

Reply:
This is a great question. We can update this field to handle IpV6 as well but this change would be different from our current behavior. What is the recommendation here?

Reply:
This is  also to keep behavior consistent. No new functionality added here.